node/bft/src/helpers/proposal_cache.rs

```diff
@@ -108,8 +106,13 @@ impl<N: Network> ProposalCache<N> {
     }
 
     /// Store the proposal cache to the file system.
-    pub fn store(&self, dev: Option<u16>) -> Result<()> {
-        let path = proposal_cache_path(N::ID, dev);
+    pub fn store(&self, storage_mode: &StorageMode) -> Result<()> {
+        // Do not write the proposal cache to the filesystem when in test mode.
+        if matches!(storage_mode, StorageMode::Test(_)) {
+            return Ok(());
```

Comment:
it's probably not needed in tests, but it technically could be stored at the temporary location (just for future reference)

Reply:
Potentially, though with the caveat that multiple test runs will result in the same file being used at the moment (as it is explicitly named for the same ID).

Reply:
@ljedrz Is the temporary test folder cleaned up in between tests / do different tests use different ones? If so, we can just store it and save ourselves the headache if we ever build a test which uses it.

Reply:
every test uses its own random temporary folder, unless specifically configured to share it (which may be useful within some specific test in the future); it's necessary in order to be able to run tests in parallel without worrying about storage clashes

Reply:
@niklaslong please fix.

Reply:
The proposal cache is only written to the fs on shutdown, I'm not convinced this is something we'd ever want in tests?

Reply:
Perhaps I got unlucky, I've tested it again with a devnet run + test runs and haven't seen the same issue without the change, so I have reverted it for now.